llvm/lib/Target/DPU/DPUTargetLowering.cpp

```diff
@@ -1803,6 +1809,42 @@ SDValue DPUTargetLowering::LowerDMA(SDValue Op, SelectionDAG &DAG,
   return DAG.getNode(DPUISD, dl, evt, Ops);
 }
 
+static uint64_t PageSizeLog2ToNcCondition(uint64_t pageSize) {
+  if (pageSize >= 10) {
+    return DPUAsmCondition::NotCarry10 + pageSize - 10;
+  } else {
+    return DPUAsmCondition::NotCarry5 + pageSize - 5;
+  }
```

Comment:
Is this function a little too much dependent on the enumerators order?

Reply:
Maybe a autogen function would be better, with a simple switch enumerating every ``NotCarryX`` condition ?

Reply:
I do not know if we really need to auto-generate this part, but ok, why not.

Reply:
I didn't use autogen.